modules/transport-netty4/src/main/java/org/elasticsearch/http/netty4/PipelinedHttpRequestPart.java

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.http.netty4;
+
+/**
+ * A part of HTTP request that has HTTP pipelining sequence number
+ */
+public sealed interface PipelinedHttpRequestPart extends PipelinedHttpObject permits PipelinedHttpRequest, PipelinedHttpContent,
+    PipelinedLastHttpContent {
```

Comment:
I'd prefer to drop this marker interface. IIUC, it is only to differentiate `PipelinedFullHttpRequest` from other `PipelinedHttpObject`. Can we instead have a method on `PipelinedHttpObject` to differentiate between them? I personally find the hierarchy a bit hard to follow.

Reply:
I planned to use this sealed interface in PartialRequest path, so I can do `switch(part)` with exhaustive check. Which is part of java 21 [jep-441](https://openjdk.org/jeps/441). But we are still compiling with java 17 and nothing fancy came out of this code. Can be removed.